test/replication-luatest/gh_7253_election_long_wal_write_test.lua

```diff
@@ -171,3 +171,86 @@ g.test_fence_during_confirm_wal_write = function(g)
     end)
     wait_fullmesh(g)
 end
+
+g.test_vote_during_txn_wal_write = function(g)
+    --
+    -- Make the following topology:
+    --   server1 <-> server2 <-> server3
+    --
+    g.server1:exec(function()
+        local replication = table.copy(box.cfg.replication)
+        rawset(_G, 'old_replication', table.copy(replication))
+        replication[3] = nil
+        box.cfg{replication = replication}
+    end)
+    g.server3:exec(function()
+        local replication = table.copy(box.cfg.replication)
+        rawset(_G, 'old_replication', table.copy(replication))
+        table.remove(replication, 1)
+        box.cfg{replication = replication}
+    end)
+    --
+    -- Server2 gets a foreign txn going to WAL too long.
+    --
+    server_block_next_wal_write(g.server2)
+    local f = fiber.new(g.server1.exec, g.server1, function()
+        box.space.test:replace({1})
+    end)
+    f:set_joinable(true)
+    server_wait_wal_is_blocked(g.server2)
+    --
+    -- Server3 tries to become a leader by requesting a vote from server2.
+    --
+    local term = g.server2:election_term()
+    fiber.create(g.server3.exec, g.server3, function()
+        box.cfg{
+            election_mode = 'manual',
+            election_timeout = 1000,
+        }
+        pcall(box.ctl.promote)
+    end)
+    g.server2:wait_election_term(term + 1)
+    --
+    -- Server2 shouldn't have persisted a vote yet. Instead, when it finishes
+    -- the txn WAL write, it sees that its vclock is > server3's one and it
+    -- cancels the vote.
+    --
+    server_unblock_wal_write(g.server2)
+    --
+    -- Server1 gets the new term via server2.
+    --
+    g.server1:wait_election_term(term + 1)
+    -- Restore server3 original params.
```

Comment:
Maybe check `box.info.election` on `server2` as well? Otherwise looks like server3 could still win and pass the test by chance

Reply:
Done.